### ISSUE TITLE
Does not work with tensorflow-gpu

### ISSUE THREAD

**Issue by Vovak1919** [AppliedDataSciencePartners/DeepReinforcementLearning#29]
After installing tensorflow-gpu, errors appeared in the model.py module:
138	def conv_layer(self, x, filters, kernel_size):

		x = Conv2D(
		filters = filters
		, kernel_size = kernel_size
		, data_format="channels_first"
		, padding = 'same'
		, use_bias=False
		, activation='linear'
		, kernel_regularizer = regularizers.l2(self.reg_const)
		)(x)

150		x = BatchNormalization(axis=1)(x)
^

ValueError
Shape must be rank 1 but is rank 0 for 'batch_normalization_1/cond/Reshape_4' (op: 'Reshape') with input shapes: [1,75,1,1], [].
  File "C:\DRL\model.py", line 150, in conv_layer
    x = BatchNormalization(axis=1)(x)
  File "C:\DRL\model.py", line 225, in _build_model
    x = self.conv_layer(main_input, self.hidden_layers[0]['filters'], self.hidden_layers[0]['kernel_size'])
  File "C:\DRL\model.py", line 114, in __init__
    self.model = self._build_model()
  File "C:\DRL\Untitled-1.py", line 67, in <module>
    current_NN = Residual_CNN(config.REG_CONST, config.LEARNING_RATE, (2,) + env.grid_shape,   env.action_size, config.HIDDEN_CNN_LAYERS)

**Comment by Vovak1919** [comment]
Shape:TensorShape([Dimension(None), Dimension(75), Dimension(6), Dimension(7)])
dims:[Dimension(None), Dimension(75), Dimension(6), Dimension(7)]
0:Dimension(None)
1:Dimension(75)
2:Dimension(6)
3:Dimension(7)
__len__:4
ndims:4
_dims:[Dimension(None), Dimension(75), Dimension(6), Dimension(7)]
_ndims:4
_tf_api_names:('TensorShape',)
_tf_api_names_v1:('TensorShape',)
value_index:0
_consumers:[]
_dtype:tf.float32
_id:21

**Comment by Vovak1919** [comment]
The module is called from here.
`# create an untrained neural network objects from the config file
current_NN = Residual_CNN(config.REG_CONST, config.LEARNING_RATE, (2,) + env.grid_shape,   env.action_size, config.HIDDEN_CNN_LAYERS)`

Call Stack:
conv_layer (c:\DRL\model.py:150)
_build_model (c:\DRL\model.py:225)
__init__ (c:\DRL\model.py:114)
<module> (c:\DRL\Untitled-1.py:67)

**Comment by Vovak1919** [comment]
There are 5 errors in the model.py module and they are all in the Gen_Model class.
Instance of 'Gen_Model' has no 'model' member (30,10); (33,10); 36,4); (42,12) and (60,12)

**Comment by Vovak1919** [comment]
Thank you, earned after correcting the file tensorflow_backend.py
[https://github.com/keras-team/keras/commit/e3a2f7d29f2f1c21ecc978bd0038b1d1330d33c2](url)

**Comment by ahmedosama9777** [comment]
What did you do as a correction?
